includes/classes/class-plugin-manager.php

```diff
@@ -0,0 +1,568 @@
+	 *
+	 * @return bool
+	 */
+	public static function can_install_plugins() {
+		if ( ( defined( 'DISALLOW_FILE_EDIT' ) && DISALLOW_FILE_EDIT ) ||
+			( defined( 'DISALLOW_FILE_MODS' ) && DISALLOW_FILE_MODS ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Activate a plugin, installing it first if necessary.
+	 *
+	 * @param string $plugin The plugin slug or URL to the plugin.
+	 * @return bool True on success. False on failure or if plugin was already activated.
+	 */
+	public static function activate( $plugin ) {
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_slug = self::get_plugin_slug( $plugin );
+		if ( ! $plugin_slug ) {
+			return new WP_Error( 'wpp_invalid_plugin', __( 'Invalid plugin.', 'performance-lab' ) );
+		}
+
+		$installed_plugins = self::get_installed_plugins();
+
+		// Install the plugin if it's not installed already.
+		$plugin_installed = isset( $installed_plugins[ $plugin_slug ] );
+		if ( ! $plugin_installed ) {
+			$plugin_installed = self::install( $plugin );
+		}
+
+		// @phpstan-ignore-next-line false flag that this will always amount to true.
+		if ( is_wp_error( $plugin_installed ) ) {
+			return $plugin_installed;
+		}
+
+		// Refresh the installed plugin list if the plugin isn't present because we just installed it.
+		if ( ! isset( $installed_plugins[ $plugin_slug ] ) ) {
+			$installed_plugins = self::get_installed_plugins();
+		}
+
+		if ( is_plugin_active( $installed_plugins[ $plugin_slug ] ) ) {
+			return new WP_Error( 'wpp_plugin_already_active', __( 'The plugin is already active.', 'performance-lab' ) );
+		}
+
+		$activated = activate_plugin( $installed_plugins[ $plugin_slug ] );
+		if ( is_wp_error( $activated ) ) {
+			return new WP_Error( 'wpp_plugin_failed_activation', $activated->get_error_message() );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deactivate a plugin.
+	 *
+	 * @param string $plugin The plugin slug (e.g. 'newspack') or path to the plugin file. e.g. ('newspack/newspack.php').
+	 * @return bool True on success. False on failure.
+	 */
+	public static function deactivate( $plugin ) {
+		$installed_plugins = self::get_installed_plugins();
+		if ( ! in_array( $plugin, $installed_plugins, true ) && ! isset( $installed_plugins[ $plugin ] ) ) {
+			return new WP_Error( 'wpp_plugin_not_installed', __( 'The plugin is not installed.', 'performance-lab' ) );
+		}
+
+		if ( isset( $installed_plugins[ $plugin ] ) ) {
+			$plugin_file = $installed_plugins[ $plugin ];
+		} else {
+			$plugin_file = $plugin;
+		}
+
+		if ( ! is_plugin_active( $plugin_file ) ) {
+			return new WP_Error( 'wpp_plugin_not_active', __( 'The plugin is not active.', 'performance-lab' ) );
+		}
+
+		deactivate_plugins( $plugin_file );
+		if ( is_plugin_active( $plugin_file ) ) {
+			return new WP_Error( 'wpp_plugin_failed_deactivation', __( 'Failed to deactivate plugin.', 'performance-lab' ) );
+		}
+		return true;
+	}
```

Comment:
Similarly to my comment on the REST classes, why do we need methods to activate and deactivate? Can't we use what core already provides for that purpose?